gk/main.c

```diff
@@ -2254,7 +2257,7 @@ gk_proc(void *arg)
 
 		process_cmds_from_mailbox(instance, gk_conf);
 
-		if (fe != NULL) {
+		if (fe != NULL && fe->in_use) {
 			gk_del_flow_entry_from_hash(
 				instance->ip_flow_hash_table, fe);
 
```

Comment:
I can't comment on the lines below the ones above, namely:
```C
if (instance->num_scan_del > 0)
		instance->num_scan_del--;
```
These lines should go into `gk_del_flow_entry_from_hash()` in order to avoid the following bug: `instance->num_scan_del` is not zero and a mailbox message flushes the whole flow table. `gk_hash_add_flow_entry()` will no longer add entries under this condition. This change is better in another patch in this same pull request.